features/netsocket/nsapi_dns.cpp

```diff
@@ -243,8 +243,9 @@ static nsapi_size_or_error_t nsapi_dns_query_multiple(NetworkStack *stack, const
         }
 
         const uint8_t *response = packet;
-        if (dns_scan_response(&response, addr, addr_count) > 0) {
-            result = NSAPI_ERROR_OK;
+        int count = dns_scan_response(&response, addr, addr_count);
```

Comment:
You might consider changing the return type of `dns_scan_response()`. It currently returns an `int`, even though its main return path is `return count;` and `count` is an `unsigned`.

Seems pretty fishy if you ask me. On the other hand, this function returns `nsapi_size_or_error_t` anyway, and that's an alias to `signed int`, so the cast from `unsigned` to `int` will happen somewhere anyway, so I guess this point is just purely stylistic.

Reply:
Yep, we've got to get from `addr_count` being unsigned ultimately back to that `nsapi_size_or_error_t`. I don't think it's worth worrying about - any changes are liable to trigger new "comparison of signed and unsigned" warning. Let's keep this change minimal and it can slip into a patch release.